### PR TITLE
First version of linked node tool

### PR DESCRIPTION
**Pull request by sebeder**
[Trello Card](https://trello.com/c/UQWuNoAd/700-kapselung-werkzeug-f%C3%BCr-verlinkenden-knoten)

This one only includes the new tool. The nodes itself are useless right now.

### CODE REVIEW

File: web/src/app/modules/views/main/editors/modules/graphical-editor/providers/properties/vertex-provider.ts [hunk 1]
```diff
@@ -56,6 +57,22 @@ export class VertexProvider extends ProviderBase {
         return vertex;
     }
 
+    public provideLinkedCEGNode(url: string, x: number, y: number, width: number, height: number, data: CEGmxModelNode): mxgraph.mxCell {
+        const value: string = null;
+        const style = this.shapeProvider.getStyle(new CEGLinkedNode());
+        const parent = this.graph.getDefaultParent();
+        this.graph.getModel().beginUpdate();
+        const vertex = this.graph.insertVertex(parent, url, value, x, y, width, height, style);
+        const l1 = this.graph.insertVertex(vertex, url + VertexProvider.ID_SUFFIX_VARIABLE, data.variable,
```
Comment (wehrlean) [inline web/src/app/modules/views/main/editors/modules/graphical-editor/providers/properties/vertex-provider.ts]: l1 is no good name

File: web/src/app/modules/views/main/editors/modules/tool-pallette/tools/ceg/ceg-linked-node-tool.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { CreateNodeToolBase } from '../create-node-tool-base';
+import { ShapeProvider } from '../../../graphical-editor/providers/properties/shape-provider';
+
+export class CEGLinkedNodeTool extends CreateNodeToolBase<CEGLinkedNode> {
+
+    protected modelType: { className: string; } = CEGModel;
+
+    public icon = 'plus';
+    public name = 'tools.addCegLinkedNode';
+    public style = new ShapeProvider(CEGModel).getStyle(CEGLinkedNode);
```
Comment (wehrlean) [inline web/src/app/modules/views/main/editors/modules/tool-pallette/tools/ceg/ceg-linked-node-tool.ts:14]: In web/src/app/modules/views/main/editors/modules/graphical-editor/providers/properties/vertex-provider.ts it is getStyle(new CEGLinkedNode()) here only getStyle(CEGLinkedNode);
Does both work? We should do it the same way.

File: web/src/app/validation/ceg/node-cycle-validator.ts [hunk 3]
```diff
@@ -14,9 +15,9 @@ type Circle = CEGConnection[];
 @Validator(CEGModel)
 export class NodeCycleValidator extends ElementValidatorBase<CEGModel> {
     // List of all CEGNodes of the model
-    private nodes: CEGNode[];
+    private nodes: (CEGNode|CEGLinkedNode)[];
```
Comment (wehrlean) [inline web/src/app/validation/ceg/node-cycle-validator.ts]: Spaces

**Review by wehrlean — CHANGES_REQUESTED** [review]
Good. Only small code-style changes

File: web/src/app/modules/views/main/editors/modules/tool-pallette/tools/ceg/ceg-linked-node-tool.ts (hunk 2 above)
Comment (sebeder) [reply web/src/app/modules/views/main/editors/modules/tool-pallette/tools/ceg/ceg-linked-node-tool.ts:14]: I changed the other place and removed the constructor call.

File: web/src/app/validation/ceg/node-cycle-validator.ts (hunk 3 above)
Comment (sebeder) [reply web/src/app/validation/ceg/node-cycle-validator.ts]: Done

File: web/src/app/modules/views/main/editors/modules/graphical-editor/providers/properties/vertex-provider.ts (hunk 1 above)
Comment (sebeder) [reply web/src/app/modules/views/main/editors/modules/graphical-editor/providers/properties/vertex-provider.ts]: I changed the name to `nameVertex`.

**Review by wehrlean — APPROVED** [review]

**Pull request merged**